core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -4636,9 +4636,11 @@ private void analyzeRowFilter(String currentIdentity, Table table, QualifiedObje
 
             ExpressionAnalysis expressionAnalysis;
             try {
-                Identity filterIdentity = Identity.forUser(filter.getIdentity())
-                        .withGroups(groupProvider.getGroups(filter.getIdentity()))
-                        .build();
+                Identity filterIdentity = filter.getIdentity()
+                        .map(filterUser -> Identity.forUser(filterUser)
+                                .withGroups(groupProvider.getGroups(filterUser))
+                                .build())
+                        .orElseGet(session::getIdentity);
```

Comment:
Can there be any difference between cases

- when ViewExpression has `empty` identity
- when ViewExpression has identity equal to current session user?

is it worth having a code comment why we don't check equality between them?

Reply:
FWIW I tried to make it so that when the expression's identity is the same as the session's user then the latter is re-used, and it was vetoed. But I think making it explicit that it should be reused is a better way to go.

Also, extending the view analogy, with views we do have special handling if the current session's user is the same as the view's owner (some restrictions are relaxed then). So there *may* be a difference between these two, but I don't think there would be any meaningful difference in semantics.